src/DurableTask.Core/FailureDetails.cs

```diff
@@ -116,7 +129,38 @@ public override string ToString()
         /// <returns>Returns <c>true</c> if the <see cref="ErrorType"/> value matches <typeparamref name="T"/>; <c>false</c> otherwise.</returns>
         public bool IsCausedBy<T>() where T : Exception
         {
+            if (string.IsNullOrEmpty(this.ErrorType))
```

Comment:
seems this code needs to be rather similar to the one in durabletask-dotnet, so I want to propose that we add a note about this

```suggestion
        public bool IsCausedBy<T>() where T : Exception
        {
            // This code is mostly a duplicate of this file: https://github.com/microsoft/durabletask-dotnet/blob/main/src/Abstractions/TaskFailureDetails.cs
            // when making changes here, especially bug fixes, please review the file above in case the changes need to be ported there as well.

            if (string.IsNullOrEmpty(this.ErrorType))
```